### PR TITLE
ICU-21268 Add error checks to avoid segmentation faults if tests run …

### PR DESCRIPTION
**Pull request by gnrunge**
…with stubdata only.

##### Checklist

- [ ] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21268
- [ ] Updated PR title and link in previous line to include Issue number
- [ ] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

**Review by sffc — DISMISSED** [review]

File: icu4c/source/test/intltest/dtfmttst.cpp [hunk 1]
```diff
@@ -3478,20 +3478,28 @@ void DateFormatTest::TestTimeZoneInLocale()
...
         SimpleDateFormat smptfmt("Z", locale, status);
+        ASSERT_OK(status);
         assertEquals("TimeZone from SimpleDateFormat constructor",
                      expectedTimezone, smptfmt.getTimeZone().getID(actual));
         assertEquals("Calendar from SimpleDateFormat constructor",
                      testLine[2], smptfmt.getCalendar()->getType());
 
         LocalPointer<DateFormat> datefmt(
                 DateFormat::createDateInstance(DateFormat::kDefault, locale));
+        if (datefmt == NULL){
```
Comment (FrankYFTang) [inline icu4c/source/test/intltest/dtfmttst.cpp:3489]: nullptr for new code please

File: icu4c/source/test/intltest/dtfmttst.cpp [hunk 2]
```diff
@@ -3478,20 +3478,28 @@ void DateFormatTest::TestTimeZoneInLocale()
...
+            return;
+        }
         assertEquals("TimeZone from DateFormat::createDateInstance",
                      expectedTimezone, datefmt->getTimeZone().getID(actual));
         assertEquals("Calendar from DateFormat::createDateInstance",
                      testLine[2], datefmt->getCalendar()->getType());
-
         LocalPointer<DateFormat> timefmt(
                 DateFormat::createTimeInstance(DateFormat::kDefault, locale));
+        if (timefmt == NULL){
```
Comment (FrankYFTang) [inline icu4c/source/test/intltest/dtfmttst.cpp:3499]: nullptr for new code please

File: icu4c/source/test/intltest/dtfmttst.cpp [hunk 3]
```diff
@@ -3500,6 +3508,10 @@ void DateFormatTest::TestTimeZoneInLocale()
         LocalPointer<DateFormat> datetimefmt(
                 DateFormat::createDateTimeInstance(
                     DateFormat::kDefault, DateFormat::kDefault, locale));
+        if (datetimefmt == NULL){
```
Comment (FrankYFTang) [inline icu4c/source/test/intltest/dtfmttst.cpp:3511]: nullptr for new code please

File: icu4c/source/test/intltest/dtfmttst.cpp (hunk 1 above)
Comment (gnrunge) [reply icu4c/source/test/intltest/dtfmttst.cpp]: Done

File: icu4c/source/test/intltest/dtfmttst.cpp (hunk 2 above)
Comment (gnrunge) [reply icu4c/source/test/intltest/dtfmttst.cpp]: Done

File: icu4c/source/test/intltest/dtfmttst.cpp (hunk 3 above)
Comment (gnrunge) [reply icu4c/source/test/intltest/dtfmttst.cpp]: Done

File: icu4c/source/test/intltest/dtfmttst.cpp [hunk 4]
```diff
@@ -3478,20 +3478,28 @@ void DateFormatTest::TestTimeZoneInLocale()
...
+        }
         assertEquals("TimeZone from DateFormat::createDateInstance",
                      expectedTimezone, datefmt->getTimeZone().getID(actual));
         assertEquals("Calendar from DateFormat::createDateInstance",
                      testLine[2], datefmt->getCalendar()->getType());
-
         LocalPointer<DateFormat> timefmt(
                 DateFormat::createTimeInstance(DateFormat::kDefault, locale));
+        if (timefmt == nullptr){
+            dataerrln("Error calling DateFormat::createDateInstance()");
```
Comment (markusicu) [inline icu4c/source/test/intltest/dtfmttst.cpp:3500]:
```suggestion
            dataerrln("Error calling DateFormat::createTimeInstance()");
```

File: icu4c/source/test/intltest/dtfmttst.cpp [hunk 5]
```diff
@@ -3478,20 +3478,28 @@ void DateFormatTest::TestTimeZoneInLocale()
...
         SimpleDateFormat smptfmt("Z", locale, status);
+        ASSERT_OK(status);
         assertEquals("TimeZone from SimpleDateFormat constructor",
                      expectedTimezone, smptfmt.getTimeZone().getID(actual));
         assertEquals("Calendar from SimpleDateFormat constructor",
                      testLine[2], smptfmt.getCalendar()->getType());
 
         LocalPointer<DateFormat> datefmt(
                 DateFormat::createDateInstance(DateFormat::kDefault, locale));
+        if (datefmt == nullptr){
```
Comment (markusicu) [inline icu4c/source/test/intltest/dtfmttst.cpp:3489]: Please use a space before `{`, here and below.

File: icu4c/source/test/intltest/dtfmttst.cpp [hunk 6]
```diff
@@ -3500,6 +3508,10 @@ void DateFormatTest::TestTimeZoneInLocale()
         LocalPointer<DateFormat> datetimefmt(
                 DateFormat::createDateTimeInstance(
                     DateFormat::kDefault, DateFormat::kDefault, locale));
+        if (datetimefmt == nullptr){
+            dataerrln("Error calling DateFormat::createDateInstance()");
```
Comment (markusicu) [inline icu4c/source/test/intltest/dtfmttst.cpp:3512]:
```suggestion
            dataerrln("Error calling DateFormat::createDateTimeInstance()");
```

**Comment by markusicu** [comment]
FYI  Feel free to use `git commit --amend -a` to keep it to a single commit.

File: icu4c/source/test/intltest/dtfmttst.cpp (hunk 4 above)
Comment (gnrunge) [reply icu4c/source/test/intltest/dtfmttst.cpp]: Done.

File: icu4c/source/test/intltest/dtfmttst.cpp (hunk 5 above)
Comment (gnrunge) [reply icu4c/source/test/intltest/dtfmttst.cpp]: Done.

File: icu4c/source/test/intltest/dtfmttst.cpp (hunk 6 above)
Comment (gnrunge) [reply icu4c/source/test/intltest/dtfmttst.cpp]: Done.

**Review by markusicu — APPROVED** [review]
lgtm tnx

**Review by FrankYFTang — APPROVED** [review]
LGTM

**Pull request merged**